src/lib/features/ppt/slideRenderers.ts

```
@@ -71,13 +92,743 @@ const LEGEND_POS_MAP: Record<string, "t" | "b" | "l" | "r"> = {
+  }
+
+  const segments: TextSegment[] = [];
+
+  // Regex to match **bold**, *italic*, or ***bold italic***
+  // Pattern: captures text between markers
+  const pattern = /(\*\*\*(.+?)\*\*\*|\*\*(.+?)\*\*|\*(.+?)\*)/g;
+
+  let lastIndex = 0;
+  let match: RegExpExecArray | null;
+
+  while ((match = pattern.exec(text)) !== null) {
```

Biome (2.3.13)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)